test/spec/Feature/Query/AggregateFunctionsSpec.hs

```diff
@@ -0,0 +1,143 @@
+      it "supports count()" $
+        get "/project_invoices?select=invoice_total.count()" `shouldRespondWith`
+          [json|[{ "count": 8 }]|] { matchHeaders = [matchContentTypeJson] }
+      it "groups by any fields selected that do not have an aggregate applied" $
+        get "/project_invoices?select=invoice_total.sum(),invoice_total.max(),invoice_total.min(),project_id&order=project_id.desc" `shouldRespondWith`
+          [json|[
+            {"sum":4100,"max":4000,"min":100,"project_id":4},
+            {"sum":3200,"max":2000,"min":1200,"project_id":3},
+            {"sum":1200,"max":700,"min":500,"project_id":2},
+            {"sum":300,"max":200,"min":100,"project_id":1} ]|]
+          { matchHeaders = [matchContentTypeJson] }
+      it "supports the use of aliases on fields that will be used in the group by" $
+        get "/project_invoices?select=invoice_total.sum(),invoice_total.max(),invoice_total.min(),pid:project_id&order=project_id.desc" `shouldRespondWith`
+          [json|[
+            {"sum":4100,"max":4000,"min":100,"pid":4},
+            {"sum":3200,"max":2000,"min":1200,"pid":3},
+            {"sum":1200,"max":700,"min":500,"pid":2},
+            {"sum":300,"max":200,"min":100,"pid":1}]|]
+          { matchHeaders = [matchContentTypeJson] }
+      it "allows you to specify an alias for the aggregate" $
+        get "/project_invoices?select=total_charged:invoice_total.sum(),project_id&order=project_id.desc" `shouldRespondWith`
+          [json|[
+             {"total_charged":4100,"project_id":4},
+             {"total_charged":3200,"project_id":3},
+             {"total_charged":1200,"project_id":2},
+             {"total_charged":300,"project_id":1}]|] { matchHeaders = [matchContentTypeJson] }
+      it "allows you to cast the result of the aggregate" $
+        get "/project_invoices?select=total_charged:invoice_total.sum()::text,project_id&order=project_id.desc" `shouldRespondWith`
+          [json|[
+             {"total_charged":"4100","project_id":4},
+             {"total_charged":"3200","project_id":3},
+             {"total_charged":"1200","project_id":2},
+             {"total_charged":"300","project_id":1}]|] { matchHeaders = [matchContentTypeJson] }
+      it "allows you to cast the input argument of the aggregate" $
+        get "/trash_details?select=jsonb_col->>key::integer.sum()" `shouldRespondWith`
+          [json|[{"sum": 24}]|] { matchHeaders = [matchContentTypeJson] }
+      it "allows the combination of an alias, a before cast, and an after cast" $
+        get "/trash_details?select=s:jsonb_col->>key::integer.sum()::text" `shouldRespondWith`
+          [json|[{"s": "24"}]|] { matchHeaders = [matchContentTypeJson] }
+      it "supports use of aggregates on RPC functions that return table values" $
+        get "/rpc/getallprojects?select=id.max()" `shouldRespondWith`
+          [json|[{"max": 5}]|] { matchHeaders = [matchContentTypeJson] }
+
+    context "performing hoisted aggregations" $ do
```

Comment:
@timabdulla I've added these tests to prove hoisting works. Now I do realize these would be better with spreading combined. But as you've noticed this doesn't work for to-many relationships.

I think we can solve that on another PR (https://github.com/PostgREST/postgrest/issues/2730 needs it too). The  output is still useful as it is.

Reply:
Tracking spreading for to-many relationships on https://github.com/PostgREST/postgrest/issues/3041